src/edu/upenn/cit594/datamanagement/DelimitedFileReader.java

```diff
@@ -0,0 +1,48 @@
+package edu.upenn.cit594.datamanagement;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.*;
+
+public abstract class DelimitedFileReader<T> extends FileReader<T> {
+    boolean hasHeaders;
```

Comment:
I think hasHeaders should be a method (something like headers != null) and the field should be the headers themselves in a List<String>.